### PR TITLE
added: cart functionality

### PR DESCRIPTION
**Pull request by rkapoor10**
added cart functionality

- implemented using  cart context and cart reducer
- check cart by the live preview link

### CODE REVIEW

**Review by neetukumari4858 — APPROVED** [review]
code is readable good to see and I like color Theam of your Website.

**Review by Raghav888 — APPROVED** [review]
Good Structure

File: src/components/cart/CartProducts.jsx [hunk 1]
```diff
@@ -1,92 +1,61 @@
...
+      <div className="space-between alignitems-c wrapit seperator-lightgray padtb-1 mb-2">
+        <h4 className="baseteal">MY CART ({cartState.totalCartItems})</h4>
+        <span className="darkteal txt-s alignitems-c">
+          <MdLocationPin /> Deliver To
+        </span>
+      </div>
+
+      <div className="cart-products flex-column-center">
+        {cartState.cart.map((product) => {
+          return (
```
Comment (JantuDeb) [inline src/components/cart/CartProducts.jsx:22]: const {id, item} = product

File: src/components/shared/navbar/NavIcons.jsx [hunk 2]
```diff
@@ -24,14 +26,18 @@ const NavIcons = () => {
...
           <FaShoppingCart
             className="navbar-icons gray icon white"
             title="Cart"
           />
-          <div className="badge-font icons-disp bg-basepink badge-top white">
-            10
+          {(cartState.totalCartItems===0)?
+            null:<div className="badge-font icons-disp bg-basepink badge-top white">
+            {cartState.totalCartItems}
           </div>
```
Comment (JantuDeb) [inline src/components/shared/navbar/NavIcons.jsx]:
```
cartState.totalCartItems===0 && <div>

```

File: src/context/cartContext/CartContext.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    totalCartPrice: 0,
+    cart: [],
+  };
+
+  const findInCart = (state, product) =>
+    state.cart.find((item) => item.id === product.id);
+
+  const cartReducer = (state, action) => {
+    console.log(state);
+    const product = action.payload;
```
Comment (JantuDeb) [inline src/context/cartContext/CartContext.jsx:20]: const {payload, type} = action

File: src/context/filterContext/FilterContext.jsx [hunk 4]
```diff
@@ -30,7 +30,7 @@ const FilterProvider = ({ children }) => {
       const { data } = await axios.get("/api/products");
       setProductsList(data.products);
     } catch (error) {
-        // yahan or better kya kr skte hai
+        
       console.log("Fetch Products Fail! in FilterContext.jsx");
```
Comment (JantuDeb) [inline src/context/filterContext/FilterContext.jsx:34]:
```
const {data:[], loading=false, error:''} = fetchProducts() 
```

**Review by JantuDeb — APPROVED** [review]
left few comments, overall these looks good

File: src/context/cartContext/CartContext.jsx (hunk 3 above)
Comment (rkapoor10) [reply src/context/cartContext/CartContext.jsx:20]: done

File: src/components/shared/navbar/NavIcons.jsx (hunk 2 above)
Comment (rkapoor10) [reply src/components/shared/navbar/NavIcons.jsx]: done

File: src/context/filterContext/FilterContext.jsx (hunk 4 above)
Comment (rkapoor10) [reply src/context/filterContext/FilterContext.jsx:34]: ok

**Pull request merged**